From dc1a3e918c7b842309d58886db6fb1b48f0f731f Mon Sep 17 00:00:00 2001
From: Pawit Pornkitprasan <p.pawit@gmail.com>
Date: Wed, 19 Nov 2014 23:22:41 +0700
Subject: [PATCH 1/2] build: clean up otasigcheck

- Only mount and unmount /data if it was originally unmounted
- Don't use comparison hack, just check the result of the script

Change-Id: I4a22485d315cf91e95ce578907c49f5fa3a03222
---
 tools/releasetools/edify_generator.py    | 4 +---
 tools/releasetools/ota_from_target_files | 5 +++++
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/tools/releasetools/edify_generator.py b/tools/releasetools/edify_generator.py
index 80077e6..2931410 100644
--- a/tools/releasetools/edify_generator.py
+++ b/tools/releasetools/edify_generator.py
@@ -167,9 +167,7 @@ class EdifyGenerator(object):
         self.script.append('package_extract_file("system/bin/otasigcheck.sh", "/tmp/otasigcheck.sh");')
         self.script.append('package_extract_file("META-INF/org/cyanogenmod/releasekey", "/tmp/releasekey");')
         self.script.append('set_metadata("/tmp/otasigcheck.sh", "uid", 0, "gid", 0, "mode", 0755);')
-        self.script.append('run_program("/tmp/otasigcheck.sh");')
-        ## Hax: a failure from run_program doesn't trigger an abort, so have it change the key value and check for "INVALID"
-        self.script.append('sha1_check(read_file("/tmp/releasekey"),"7241e92725436afc79389d4fc2333a2aa8c20230") && abort("Can\'t install this package on top of incompatible data. Please try another package or run a factory reset");')
+        self.script.append('run_program("/tmp/otasigcheck.sh") == "0" || abort("Can\'t install this package on top of incompatible data. Please try another package or run a factory reset");')
 
   def ShowProgress(self, frac, dur):
     """Update the progress bar, advancing it over 'frac' over the next
diff --git a/tools/releasetools/ota_from_target_files b/tools/releasetools/ota_from_target_files
index d3787d2..d036c43 100755
--- a/tools/releasetools/ota_from_target_files
+++ b/tools/releasetools/ota_from_target_files
@@ -589,9 +589,14 @@ else if get_stage("%(bcb_dev)s") == "3/3" then
   if block_based:
     common.ZipWriteStr(output_zip, "system/bin/otasigcheck.sh",
                    ""+input_zip.read("SYSTEM/bin/otasigcheck.sh"))
+
+  script.AppendExtra("if is_mounted(\"/data\") then")
+  script.ValidateSignatures("data")
+  script.AppendExtra("else")
   script.Mount("/data")
   script.ValidateSignatures("data")
   script.Unmount("/data")
+  script.AppendExtra("endif;")
 
   if "selinux_fc" in OPTIONS.info_dict:
     WritePolicyConfig(OPTIONS.info_dict["selinux_fc"], output_zip)
-- 
1.9.3 (Apple Git-50)


From bc097e592db7c1bbc5e2be37ce6c3b1d9345fe21 Mon Sep 17 00:00:00 2001
From: Pawit Pornkitprasan <p.pawit@gmail.com>
Date: Wed, 19 Nov 2014 23:21:37 +0700
Subject: [PATCH 2/2] i9082: HACK: add cortex-a9 mcpu flag

(waiting for CM patch to include all flags)

Change-Id: If52acb8345f7d1d29589df1d0718b3f8caf945c6
---
 core/combo/arch/arm/armv7-a-neon.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/combo/arch/arm/armv7-a-neon.mk b/core/combo/arch/arm/armv7-a-neon.mk
index a7185f0..baed1a8 100644
--- a/core/combo/arch/arm/armv7-a-neon.mk
+++ b/core/combo/arch/arm/armv7-a-neon.mk
@@ -20,6 +20,10 @@ endif
 endif
 endif
 
+ifeq ($(strip $(TARGET_$(combo_2nd_arch_prefix)CPU_VARIANT)),cortex-a9)
+	arch_variant_cflags := -mcpu=cortex-a9
+endif
+
 arch_variant_cflags += \
     -mfloat-abi=softfp \
     -mfpu=neon
-- 
1.9.3 (Apple Git-50)

